File: backend/HotelBookingEngine.Api/Features/Rooms/Room.cs
namespace HotelBookingEngine.Api.Features.Rooms;

public class Room
{
    public int Id { get; set; }
    public int RoomTypeId { get; set; }
    public int HotelId { get; set; }
    public required string RoomNumber { get; set; }
    public RoomStatus Status { get; set; }
}

File: frontend/src/types/room.ts
export type RoomStatus = "Available" | "Maintenance";

export interface Room {
  id: number;
  roomTypeId: number;
  hotelId: number;
  roomNumber: string;
  status: RoomStatus;
}

export interface RoomRequest {
  roomNumber: string;
  status: RoomStatus;
}
